app/registries/foreman/setting_manager.rb

```diff
@@ -37,25 +37,32 @@ def storage
         SettingManager.settings
       end
 
+      def available_types
+        [:boolean, :integer, :float, :string, :text, :hash, :array]
+      end
+
       # Adds setting definition
       #
       # ===== Example
       #
       #   SettingManager.define(:puppet) do
       #     category(:cfgmgmt, N_('Configuration Management')) do
       #       setting(:use_cooler_puppet,
+      #               type: :boolean,
       #               default: true,
       #               description: N_('Use Puppet that goes to 11'),
       #               full_name: N_('Use shiny puppet'),
       #               encrypt: true)
       #     end
       #   end
       #
-      def setting(name, default:, description:, full_name: nil, value: nil, collection: nil, encrypted: false, **options)
+      def setting(name, default:, description:, type:, full_name: nil, value: nil, collection: nil, encrypted: false, **options)
         raise ::Foreman::Exception.new(N_("Setting '%s' is already defined, please avoid collisions"), name) if storage.key?(name.to_s)
+        raise ::Foreman::Exception.new(N_("Setting '%s' has invalid type definition. Please use valid type."), name) unless available_types.include?(type)
```

Comment:
Is this something a user would ever see? I'm not sure if this needs to be translatable.
```suggestion
        raise ::Foreman::Exception.new(N_("Setting '%s' has an invalid type definition. Please use a valid type."), name) unless available_types.include?(type)
```